features/include/pcl/features/gasd.h

```diff
@@ -0,0 +1,252 @@
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#ifndef PCL_FEATURES_GASD_H_
+#define PCL_FEATURES_GASD_H_
+
+#include <pcl/features/feature.h>
+#include <pcl/common/common.h>
+#include <pcl/point_cloud.h>
+
+namespace pcl
+{
+  /** \brief GASDEstimation estimates the Globally Aligned Spatial Distribution (GASD) descriptor for a given
+   * point cloud dataset given XYZ data (and optionally RGB data).
+   *
+   * The suggested PointOutT is pcl::GASDSignature512 for shape only descriptor and pcl::GASDSignature984 for shape + color descriptor.
+   *
+   * \note If you use this code in any academic work, please cite:
+   *
+   *   - J. Lima, V. Teichrieb.
+   *     An Efficient Global Point Cloud Descriptor for Object Recognition and Pose Estimation.
+   *     In Proceedings of the 29th SIBGRAPI - Conference on Graphics, Patterns and Images,
+   *     Sao Jose dos Campos, Brazil, October 4-7 2016.
+   *
+   * \author Joao Paulo Lima
+   *
+   * Voxar Labs, Centro de Informatica, Universidade Federal de Pernambuco, Brazil
+   *
+   * Departamento de Estatistica e Informatica, Universidade Federal Rural de Pernambuco, Brazil
+   *
+   * \ingroup features
+   */
+  template <typename PointInT, typename PointOutT = GASDSignature984>
+  class GASDEstimation : public Feature<PointInT, PointOutT>
+  {
+    public:
+      typedef boost::shared_ptr<GASDEstimation<PointOutT> > Ptr;
+      typedef boost::shared_ptr<const GASDEstimation<PointOutT> > ConstPtr;
+
+      using Feature<PointInT, PointOutT>::feature_name_;
+      using Feature<PointInT, PointOutT>::getClassName;
+      using Feature<PointInT, PointOutT>::indices_;
+      using Feature<PointInT, PointOutT>::k_;
+      using Feature<PointInT, PointOutT>::search_radius_;
+      using Feature<PointInT, PointOutT>::surface_;
+
+      typedef typename Feature<PointInT, PointOutT>::PointCloudOut PointCloudOut;
+      typedef typename pcl::PointCloud<PointInT>::Ptr PointInTPtr;
+
+      /** \brief Different histogram interpolation methods.
+       * <ul>
+       *   <li><b>INTERP_NONE</b> - no interpolation.</li>
+       *   <li><b>INTERP_TRILINEAR</b> - trilinear interpolation.
+       *   <li><b>INTERP_QUADRILINEAR</b> - quadrilinear interpolation.
+       * </ul>
+       */
+      enum HistogramInterpolationMethod
+      {
+        INTERP_NONE, INTERP_TRILINEAR, INTERP_QUADRILINEAR
+      };
+
+      /** \brief Empty constructor. */
+      GASDEstimation () :
+          view_direction_ (0.0f, 0.0f, 1.0f),
+          shape_half_grid_size_ (3),
+          shape_hists_size_ (1),
+          color_half_grid_size_ (2),
+          color_hists_size_ (12),
+          use_color_ (true),
+          shape_interp_ (INTERP_NONE),
+          color_interp_ (INTERP_NONE)
```

Comment:
In terms of API use it will still allow you to use a constructor without explicitly passing any arguments to it. It simply gives you more flexibility.